pkg/tracing/traceLogger.go

```diff
@@ -201,9 +201,14 @@ func (l traceLogger) Error(err error, attrs ...attribute.KeyValue) error {
 		trace.WithAttributes(attrs...),
 	}
 
-	if st := findStack(err); st != nil {
```

Comment:
👀 Careful review is needed here, as it's a non-trivial update. I have replaced the `pkg/errors` search for the closest error in the stack with recursive calls to `errors.Unwrap`.

Reply:
Okay, I think this is the origin of one failing integration test [here](https://dev.azure.com/getporter/porter/_build/results?buildId=4588&view=logs&j=c892508d-390a-5765-0ad0-b341c7fd0dd4&t=99b6391b-efca-5f28-e88f-b44afecfc894&l=10423):
```
=== RUN   TestPlugins_CatchStderr/plugin_throws_an_error
    pluggable_integration_test.go:59: 
        	Error Trace:	pluggable_integration_test.go:59
        	Error:      	Error message not equal:
        	            	expected: "could not connect to the secrets.testplugin.vault plugin: Unrecognized remote plugin message: \n\nThis usually means that the plugin is either invalid or simply\nneeds to be recompiled to support the latest protocol.: i am just test plugin. i don't function\n"
        	            	actual  : "Unrecognized remote plugin message: \n\nThis usually means that the plugin is either invalid or simply\nneeds to be recompiled to support the latest protocol."
        	Test:       	TestPlugins_CatchStderr/plugin_throws_an_error
```

We Unwrap the error differently and the error message is not the same anymore. The new one is less useful. We are failing [here](https://github.com/getporter/porter/blob/afff47f90cd0e8e08fd06db610304fc8e264813a/pkg/plugins/pluggable/connection.go#L157-L165), and we end up displaying the default error message returned by `go-plugin` which is [here](https://github.com/hashicorp/go-plugin/blob/5dee41c45dcec4c3b00c6d227bec8e5ebea56a8a/client.go#L697-L702).

So we have to fix what I did here and format the error message correctly, like before I guess. I'm not sure how I will do that yet, I'll try to debug that on a playground to see how to Unwrap properly errors "recursively".

Reply:
This change in behavior is because of another PR that recently went in (#2185) that changed how the error you are handling is generated. You're right we don't need the extra logic anymore to unwrap the root error and can remove the commented out unwrap calls.